### PR TITLE
Remove Form--horizontal and refactor

### LINKED ISSUE #1: Make use of Button component
Think it might be worth reusing that component here?

### PR DESCRIPTION
**Pull request by giuseppeg**
Fixes #1 and #3

### CODE REVIEW

File: lib/form.css [hunk 1]
```diff
@@ -126,38 +120,39 @@
...
+          var(
+            --Form-select-background
+          )
+        ) 1x,
+        url(
+          var(
+            --Form-select-background-2x
+          )
+        ) 2x
+      );
```
Comment (simonsmith) [inline lib/form.css]: Maybe easier to use SVG?

**Comment by giuseppeg** [comment]
@simonsmith I was thinking of adding a modifier for the `Form-field` to get overlabels like this one http://codepen.io/giuseppegurgone/pen/BjrGXB Do you think that it may be useful?

**Comment by simonsmith** [comment]
It's cool, but maybe a little too specific to go into a generic form component. Feels like `components-form` should focus on common patterns.

**Comment by giuseppeg** [comment]
@simonsmith what you think about the latest revision?

**Comment by simonsmith** [comment]
Less code ftw 👍

**Pull request merged**